### PR TITLE
initial lab files

### PR DESCRIPTION
**Pull request by NickR2600**
Adding lab files for first 5 lessons

### CODE REVIEW

File: cmake-docker-environment/labs/cmake-example/greeting.h [hunk 1]
```diff
@@ -0,0 +1 @@
+void sayHello();
```
Comment (memsharded) [inline cmake-docker-environment/labs/cmake-example/greeting.h]: At least add #pragma once

File: cmake-docker-environment/labs/cmake-example/readme.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+   $ cmake ..
+   $ make
+   $ ./greeting
+   ```
+
+## Exercise 2: Examine the flexibility of CMake's MakeFile file
+Make a slight change to greeting.cpp and build again.
+   ```
+   $ make
+   $ ./greetingTest
```
Comment (memsharded) [inline cmake-docker-environment/labs/cmake-example/readme.md]: why the executable name has changed? It seems the exercise is only changing the c++ code, not the CMakeLists.txt

File: cmake-docker-environment/labs/consume-shared-library/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.9.1)
+project(contumerTest)
+
+add_library(greeting greeting/greeting.cpp)
+target_include_directories(greeting PUBLIC ${CMAKE_CURRENT_LIST_DIR}/greeting/include)
+
+add_executable(consumer main.cpp)
+target_link_libraries(consumer greeting)
```
Comment (memsharded) [inline cmake-docker-environment/labs/consume-shared-library/CMakeLists.txt]: All text files must end with LF

File: cmake-docker-environment/labs/consume-shared-library/greeting/include/greeting.h [hunk 4]
```diff
@@ -0,0 +1 @@
+void sayHello();
```
Comment (memsharded) [inline cmake-docker-environment/labs/consume-shared-library/greeting/include/greeting.h]: Same ``#pragma once``

File: cmake-docker-environment/labs/consume-shared-library/readme.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
+# consume-shared-library
+Commands for generating a SHARED library and consume it from an executable. Before you start, change to the proper directory and make the build directory.
+   ```
+   $ cd consume-shared-library
+   $ mkdir build && cd build
+   ```
+   
+## Exercise 1: Generate and consume a STATIC libary, check its size
+   ```
+   $ cmake -D BUILD_SHARED_LIBS=true ..
```
Comment (memsharded) [inline cmake-docker-environment/labs/consume-shared-library/readme.md]: Better use ``BUILD_SHARED_LIBS=ON``, it is more common

File: cmake-docker-environment/labs/consume-shared-library/readme.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+   void sayHello() {
+	   std::cout << "Hola World!\n";
+   }
+   ```
+
+**CMakeLists.txt**
+   ```
+   cmake_minimum_required(VERSION 3.9.1)
+   project(contumerTest)
```
Comment (memsharded) [inline cmake-docker-environment/labs/consume-shared-library/readme.md]:
```suggestion
   project(consumerTest)
```

File: cmake-docker-environment/labs/consume-static-library/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.9.1)
+project(contumerTest)
```
Comment (memsharded) [inline cmake-docker-environment/labs/consume-static-library/CMakeLists.txt]:
```suggestion
project(consumerTest)
```

**Review by memsharded — COMMENTED** [review]
I think a better layout would be to have:
- docker 
    - Dockerfile
    - docker-compose.yml
- labs
    - ....

File: cmake-docker-environment/labs/consume-shared-library/readme.md (hunk 5 above)
Comment (NickR2600) [reply cmake-docker-environment/labs/consume-shared-library/readme.md]: Updated in new commit.

File: cmake-docker-environment/labs/consume-shared-library/greeting/include/greeting.h (hunk 4 above)
Comment (NickR2600) [reply cmake-docker-environment/labs/consume-shared-library/greeting/include/greeting.h]: Fixed in later commit.

File: cmake-docker-environment/labs/cmake-example/greeting.h (hunk 1 above)
Comment (NickR2600) [reply cmake-docker-environment/labs/cmake-example/greeting.h]: Fixed in later commit

File: cmake-docker-environment/labs/cmake-example/readme.md (hunk 2 above)
Comment (NickR2600) [reply cmake-docker-environment/labs/cmake-example/readme.md]: I've fixed line 17 in a later commit to say ./greeting. Correct, the goal of this exercise is to show how the make file created by cmake offers the same time-saving benefits as the hand-written make file by recompiling only files that have changed. Later exercises will ask students to modify an existing CMakeLists.txt or write a CMakeLists.txt file from scratch.

File: cmake-docker-environment/labs/consume-shared-library/CMakeLists.txt (hunk 3 above)
Comment (NickR2600) [reply cmake-docker-environment/labs/consume-shared-library/CMakeLists.txt]: How do I fix this?

**Comment by NickR2600** [comment]
How important is revising the directory layout? It would require testing and some non-trivial edits to the lesson2 video.

File: cmake-docker-environment/labs/consume-shared-library/CMakeLists.txt (hunk 3 above)
Comment (memsharded) [reply cmake-docker-environment/labs/consume-shared-library/CMakeLists.txt]: Just make sure in your editor that files have a last blank line

**Comment by memsharded** [comment]
> How important is revising the directory layout? It would require testing and some non-trivial edits to the lesson2 video.

Not very important, not necessary to do if it requires a relevant effort, it is just a "nice to have".

**Comment by NickR2600** [comment]
Ok. I believe I've included all the other edits. Does everything look up to par?

**Pull request merged**